geopandas/tests/test_crs.py

```diff
@@ -336,6 +336,44 @@ def test_dataframe_setitem(self):
         assert df["geometry"].crs == self.wgs
         assert df["other_geom"].crs == self.osgb
 
+    def test_dataframe_multiindex_cols_2level(self):
+        # GH1763 https://github.com/geopandas/geopandas/issues/1763
+        df = pd.DataFrame(
+            [[1, 0], [0, 1]], columns=[["location", "location"], ["x", "y"]]
+        )
+        x_col = df["location", "x"]
+        y_col = df["location", "y"]
+
+        gdf = GeoDataFrame(df, crs=self.wgs, geometry=points_from_xy(x_col, y_col))
+        assert gdf.crs == self.wgs
+        assert gdf.geometry.crs == self.wgs
+        assert gdf.geometry.dtype == "geometry"
+        assert gdf._geometry_column_name == ("geometry", "")
+        assert gdf.geometry.name == ("geometry", "")  # or is "geometry" this expected?
```

Comment:
Not sure whether this is desirable. It means there is a disparity between `gdf.geometry.name` and `gdf['geometry'].name (but `gdf.geometry.name` is the same as `gdf[("geometry", "")].name

Reply:
I think it is expected, we're looking for a position of the active geom column in the index.